analyzers/tests/SonarAnalyzer.UnitTest/Metrics/CSharpExecutableLinesMetricTest.cs

```diff
@@ -627,6 +627,205 @@ public void Foo(bool flag)
                 }",
                 4, 6);
 
+        [TestMethod]
+        public void MultiLineLoop() =>
+            AssertLinesOfCode(
+              @"
+                        void Foo(int[] arr)
+                        {
+                            for         // +1
+                              (         // +0
+                               int i=0; // +0
+                               i < 10;  // +0
+                               i++      // +0
+                              )         // +0
+                            {           // +0
+                            }
+                        }",
+              4);
+
+        [TestMethod]
+        public void SwitchStatementWithMultipleCases() =>
+            AssertLinesOfCode(
+              @"
+                        void Foo(int? i, string s)
+                        {
+                            switch (i) // +1
+                            {
+                                case 1:
+                                    Console.WriteLine(4); // +1
+                                    break; // +1
+                                case 2:
+                                    Console.WriteLine(4); // +1
+                                    break; // +1
+                                default:
+                                    break; // +1
+                            }
+                        }",
+              4, 7, 8, 10, 11, 13);
+
+        [TestMethod]
+        public void SwitchExpressionWithMultipleCases() =>
+            AssertLinesOfCode(
+              @"
+                        void Foo(int? i, string s)
+                        {
+                            var x = s switch
+                            {
+                                    ""a"" => true, // FN
+                                    ""b"" => false, // FN
+                                    _ => true // FN
+                            };
+                            var y = s switch
+                            {
+                                ""a"" => Foo(""b""), // +1
+                                ""b"" => Foo(""a""), // +1
+                                _ => false // FN
+                            };
+                        }
+                        bool Foo(string s) => true;
+                ",
+              12, 13);
+
+        [TestMethod]
+        public void MultiLineInterpolatedString() =>
```

Comment:
could you please also add a test with a multi-line interpolated string which has some complex expressions on each line e.g.

```
public class C {
    public string M(int i) {
        return @$"
        {(i == 1 ? Bar(1) : Bar(2))}
        {(i == 2 ? Bar(2) : Bar(3))}
        {Bar(5)}
            ";
    }
    
    public string Bar(int i) => "y" + i;
}
```

The interpolated string is in a single statement so it should be counted only once

> Regardless of the number of lines across which nested statements are spread, the executable line count should only be incremented by one, since typically the execution of one naturally follows from the other.